src/client-scopes/details/MappingDetails.tsx

```diff
@@ -71,11 +75,17 @@ export const MappingDetails = () => {
         };
       } else {
         const scope = await adminClient.clientScopes.findOne({ id });
+        if (!scope) {
+          throw new Error(t("common:notFound"));
+        }
         const protocolMappers =
           serverInfo.protocolMapperTypes![scope.protocol!];
         const mapping = protocolMappers.find(
           (mapper) => mapper.id === mapperId
         )!;
```

Comment:
Since this value is now checked this non-null assertion can be removed.
```suggestion
        );
```